src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx

```diff
@@ -660,7 +660,36 @@ const getScoreConditions = (items?: Item[], conditions?: Condition[]) =>
     return {
       ...condition,
       payload,
-      itemName: relatedItem ? getEntityKey(relatedItem) : condition.itemName,
+      itemName: relatedItem ? getEntityKey(relatedItem) : scoreName ?? condition.itemName,
+    };
+  });
+
+const findRelatedScore = (itemName: string, scores?: ScoreReport[]) => {
```

Comment:
could you pls reuse similar functions in SaveAndPublish.utils.ts and BuilderApplet.utils.tsx to not duplicate the code?

Reply:
done